### PR TITLE
feat: reconnect socket automatically on close event

### PR DESCRIPTION
**Pull request by lukeocodes**

### CODE REVIEW

File: src/packages/LiveClient.ts [hunk 1]
```diff
@@ -16,33 +16,62 @@ import type {
...
+  }
 
-    this._socket = new w3cwebsocket(url.toString(), ["token", this.key]);
+  protected _attemptReconnection(): void {
+    if (this._reconnections > WS_RECONNECTION_RETRY_LIMIT) {
+      throw new DeepgramError("Reconnection retry limit exceeded");
+    }
+
+    this._reconnections += 1;
+    this._unregisterEvents();
```
Comment (dvonthenen) [inline src/packages/LiveClient.ts:52]: Would the user be interested in getting the `onopen` event to know that the WS successfully reconnected?

File: test/live.test.ts [hunk 2]
```diff
@@ -7,9 +7,9 @@ import { LiveConnectionState, LiveTranscriptionEvents } from "../src/lib/enums";
 describe("connecting to our transcription websocket", () => {
   let deepgram: DeepgramClient;
 
-  beforeEach(() => {
+  before(() => {
     deepgram = createClient(faker.string.alphanumeric(40), {
-      global: { url: "https://api.mock.deepgram.com" },
+      global: { url: "https://api.mock.deepgram" },
```
Comment (dvonthenen) [inline test/live.test.ts:12]: Not sure if the removal of `.com` is intentional.

**Review by dvonthenen — COMMENTED** [review]
Just checking to see if these are desired or not.

File: test/live.test.ts (hunk 2 above)
Comment (lukeocodes) [reply test/live.test.ts:12]: It is intentional! But the commit of it wasn't lol. thanks!

Comment (lukeocodes) [reply test/live.test.ts:12]: fixed

File: src/packages/LiveClient.ts (hunk 1 above)
Comment (lukeocodes) [reply src/packages/LiveClient.ts:52]: We reregister the events after. These internal events emit an external one, which won't need to be reregistered. So they're likely to get second onopen event... which may be troublesome for some actually

**Comment by lukeocodes** [comment]
I need to get the key factory in and documented before this, and I may change this to be a method users can call rather than automatically reconnect

**Pull request closed without merging**